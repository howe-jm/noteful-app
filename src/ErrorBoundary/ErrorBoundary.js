import React from 'react';
import PropTypes from 'prop-types';

export default class ErrorPage extends React.Component {
  state = { error: null };
  static getDerivedStateFromError(error) {
    console.error(error);
    return { error };
  }
  render() {
    if (this.state.error) {
      return (
        <main className='error-page'>
          <h1>Something seems to have gone wrong</h1>
          <p>Try refreshing the page</p>
        </main>
      );
    }
    return this.props.children;
  }
}

ErrorPage.defaultProps = {
  children: PropTypes.obj,
};
