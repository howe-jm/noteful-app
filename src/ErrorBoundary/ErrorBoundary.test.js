import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorBoundary from './ErrorBoundary';

describe(`Add Folder component`, () => {
  it('Renders without errors', () => {
    const wrapper = shallow(<ErrorBoundary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
