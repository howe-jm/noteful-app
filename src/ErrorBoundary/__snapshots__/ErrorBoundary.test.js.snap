// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Folder component Renders without errors 1`] = `""`;
